packages/go_router/lib/src/router.dart

```diff
@@ -121,6 +122,7 @@ class GoRouter implements RouterConfig<RouteMatchList> {
   /// The `routes` must not be null and must contain an [GoRouter] to match `/`.
   factory GoRouter({
     required List<RouteBase> routes,
+    Codec<Object?, Object?>? extraCodec,
```

Comment:
I have been debating what is the best format to provide codec for multiple class type since GoRouter does not limit how many types can be use as extra during its life cycle.

Things I have tried: 
1. A list<Codec>
There isn't a way to get what type the codec support, so it is not possible to choose a codec for a given class type
2. A Map<Type, Codec>
This is fine when encoding the data, but I can't find a good way to get the codec for decoded data.


I end up go with this approach where it would rely on the developer to check the type in their codec and write certain marker in the encoded result to be picked up during decoding